packages/web/src/components/search-bar/SearchBarResult.tsx

```diff
@@ -1,38 +1,82 @@
+import { useCollection, useTrack, useUser } from '@audius/common/api'
 import { imageBlank, imageProfilePicEmpty } from '@audius/common/assets'
 import {
   User,
   UserTrackMetadata,
-  UserCollectionMetadata
+  UserCollectionMetadata,
+  ID,
+  Kind
 } from '@audius/common/models'
+import { SearchItemBackwardsCompatible } from '@audius/common/src/store/search/types'
+import { searchActions } from '@audius/common/store'
 import { route } from '@audius/common/utils'
-import { Text, Flex, Avatar, Artwork } from '@audius/harmony'
+import { Text, Flex, Avatar, Artwork, IconCloseAlt } from '@audius/harmony'
+import { useDispatch } from 'react-redux'
 import { Link } from 'react-router-dom'
+import { useNavigate } from 'react-router-dom-v5-compat'
 
 import UserBadges from 'components/user-badges/UserBadges'
 
 import styles from './DesktopSearchBar.module.css'
 
 const { profilePage, collectionPage } = route
+const { addItem: addRecentSearch } = searchActions
 
 const ResultWrapper = ({
   children,
-  to
+  to,
+  onRemove,
+  kind,
+  id
 }: {
   children: React.ReactNode
   to: string
-}) => (
-  <Flex
-    as={Link}
-    // @ts-expect-error
-    to={to}
-    alignItems='center'
-    gap='s'
-    p='s'
-    css={{ minWidth: 0 }}
-  >
-    {children}
-  </Flex>
-)
+  onRemove?: () => void
+  kind: Kind
+  id: ID
+}) => {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  return (
+    <Flex
+      alignItems='center'
+      justifyContent='space-between'
+      p='s'
+      css={{
+        minWidth: 0
+      }}
+    >
+      <Flex
+        as={Link}
+        gap='s'
+        alignItems='center'
+        css={{ minWidth: 0 }}
+        onClick={() => {
+          dispatch(
+            addRecentSearch({
```

Comment:
big nit, but can likely put the addRecentSearch fn data all in one or two lines, prettier will listen